Feature/maps user profile

haideralishah opened the pull request:
- [x] Back button is often unresponsive in the Maps/ Single Contribution Page, Improve the button performance.

- [x] Clicking on tree icon on map, often takes to wrong/different contribution. This has been fixed for navigation from cards, but not from icons.

- [x] Loading a single contribution fetches all projects related to contributions. instead it should fetch neighbors only When contribution cards are stacked as `A [B] C` B being the current view, only load projects related to A, B & C, (performance)

- [x] Remove Current Location Icon from the Map View

- [x] Change to a icon of a tree for map marker instead of image (performance)

- [ ] Use Apple Maps in iOS ( if it is more performant than google maps)
- [x] disable the marker press if detail not show
- [x] revert the `headerText` on outer card list
- [x] disable arrows keys on both ends 
- [x] clicking on multiple card on outer list
- [x] Indexing issues
- [x] In Single Contribution, enable Satellite View by default.

Things to consider:

- [x] Look into Map Clustering




------------
Fixes #1958

Changes in this pull request:
- Show road map instead of satellite
- Center Map so that more than one items are visible.

@sagararyal

norbertschuler commented on app/components/UserHome/FullMapComponent.js:
```diff
@@ -0,0 +1,548 @@
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import MapView from 'react-native-maps';
+import { tree_1 } from '../../assets/index';
+import Smalltreewhite from '../../assets/images/smalltreewhite.png';
```
Is there any special reason you don't load this asset from `assets/index` as e.g. `tree_1`?

haideralishah replied:
I used this image 
https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/397e98913da9f1cf23fa51e7705d64cd4e7d9823/app/components/UserHome/FullMapComponent.js#L248

norbertschuler replied:
I meant why do you directly import `Smalltreewhite` and not put this image into  https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/app/assets/index.js?

norbertschuler replied:
I meant why do you directly import `Smalltreewhite` and not put this image into  https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/app/assets/index.js?

haideralishah replied:
fixed

norbertschuler commented on app/components/UserHome/FullMapComponent.js:
```diff
@@ -0,0 +1,548 @@
+        </View>
+        <View style={{ marginVertical: 5 }}>
+          <Text style={styles.subHeaderText}>
+            Planted by Eden Reforestation Projects
```
Is this still work in progress with the sample text elements not using text resources?

haideralishah replied:
I need `Heading` and `subheading` data in single tree contribution API. I already requested @harshvitra, @ankitecd, and @jmiridis to add this data in API so I will replace it with dynamic data.
no need to use text resources

sagararyal commented:
- [ ] if markers in different part of the world, zoom out to show all markers

<img width="468" alt="Screen Shot 2020-03-02 at 12 38 21 PM" src="https://user-images.githubusercontent.com/5401575/75673214-b6c3a380-5c82-11ea-8854-c9e758fe4049.png">

- [ ] Button is not circle

<img width="423" alt="Screen Shot 2020-03-02 at 12 34 30 PM" src="https://user-images.githubusercontent.com/5401575/75673014-4583f080-5c82-11ea-95df-2f5539a9f9cc.png">

- [ ] clicking full screen error

<img width="472" alt="Screen Shot 2020-03-02 at 12 35 03 PM" src="https://user-images.githubusercontent.com/5401575/75673027-4ae13b00-5c82-11ea-9f5f-42ae3c454dd7.png">


- [ ] Card Sizes are too big
<img width="467" alt="Screen Shot 2020-03-02 at 12 39 22 PM" src="https://user-images.githubusercontent.com/5401575/75673265-d955bc80-5c82-11ea-995d-07a40f011f60.png">

- [ ] Remove the while space at the bottom like at top
<img width="439" alt="Screen Shot 2020-03-02 at 12 40 07 PM" src="https://user-images.githubusercontent.com/5401575/75673315-f4c0c780-5c82-11ea-8546-bd13601837f8.png">

- [ ] Clicking a marker does not always center the map

- [ ] Use the map marker as in design

- [ ] Use Google Maps style in full screen; do not load a new layer.

- [ ] Hide button animation doesn't work as a user would expect
<img width="471" alt="Screen Shot 2020-03-02 at 12 43 14 PM" src="https://user-images.githubusercontent.com/5401575/75673528-73b60000-5c83-11ea-8d0f-f0a371d7ee1d.png">

- [ ] Add space above the button  for phones that use safe area view.

haideralishah commented:
- Google Map Style doesn't work on Apple Maps.
- I change the Icon `down arrow` to `close`
- Combine with Single Contribution Page
currently, when users tap on any card it's navigating the single contribution detail page.

haideralishah commented:
Hide button animation doesn't work as a user would expect Done.
![Hnet-image](https://user-images.githubusercontent.com/9020813/75876686-fb447000-5e37-11ea-931d-2553d65fa7d5.gif)

haideralishah commented:
Google Map Style is now working fine on iOS.
![image](https://user-images.githubusercontent.com/9020813/75877120-c8e74280-5e38-11ea-8507-fc5dc711f4ee.png)

haideralishah commented:
Combine with Single Contribution Page
![Hnet-image (1)](https://user-images.githubusercontent.com/9020813/75877909-7ad33e80-5e3a-11ea-8c79-fc77faa2bf4c.gif)

norbertschuler commented on app/components/UserHome/FullMapComponent.js:
```diff
@@ -0,0 +1,554 @@
+          longitude: -122.4324,
+          latitudeDelta: 0.0922,
+          longitudeDelta: 0.0421
+        }
```
This is somewhere in San Fransisco. @sagararyal what should be default coordinates to use in the app?

harshvitra commented:
Bottom Navigation hides when user visits the Map and comes back to the Me Page

norbertschuler commented on ios/TreecounterApp.xcodeproj/project.pbxproj:
```diff
@@ -30,12 +30,12 @@
 		891FA4A6378545F783E5B6A9 /* OpenSans-Light.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 4F0548F411A546C79D230135 /* OpenSans-Light.ttf */; };
 		8E91A3D298D54668A81191C0 /* OpenSans-ExtraBold.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 16506A78B0404858BF1334D1 /* OpenSans-ExtraBold.ttf */; };
 		96E3C6F17CE24BD5AFF45E7B /* OpenSans-Bold.ttf in Resources */ = {isa = PBXBuildFile; fileRef = B0362431F9D24179A7E9175E /* OpenSans-Bold.ttf */; };
-		ABCD2BA66D672482E2F5A002 /* libPods-TreecounterApp.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 34D4B68F0F12DDDBB81E2F5B /* libPods-TreecounterApp.a */; };
```
We should just copy the file from the develop branch into this branch, so we can avoid merge back any unnecessary changes to this file. Probably it changed due to rebuilding the iOS project (or we changed it in the develop branch).

norbertschuler replied:
Did it myself.

norbertschuler commented on app/utils/getRandomImage.js:
```diff
@@ -11,8 +11,9 @@ import {
 } from '../assets';
 
 export default function GetRandomImage(props) {
+  console.log(props, 'propsprops');
```
Please remove this or replace with `debug()` if you want to merge it.

norbertschuler replied:
cleanup up myself

norbertschuler commented:
- [ ] design issue showing my trees very much at the borders - maybe zoom out a little bit:

![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 24 48](https://user-images.githubusercontent.com/1532418/77073707-3d69d600-6a22-11ea-8503-978bcb67bd00.png)

norbertschuler commented:
Issues in user contribution detail screen:

- [ ] Why is there a line break though it seems there is enough space for the name?
- [ ] Nothing happens when I click the project name 

![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 27 47](https://user-images.githubusercontent.com/1532418/77073794-65f1d000-6a22-11ea-959c-dd9b55af574e.png)

- [ ] When you use the arrow to change the map to fullscreen, the name of the user contribution changes to the first user contribution in my list instead of the currently opened contribution.
- [ ] The name of the user contribution can be clicked, but nothing happens.

![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 39 25](https://user-images.githubusercontent.com/1532418/77074194-08aa4e80-6a23-11ea-97d2-e41a612417bc.png)

- [ ] Switching between full map and partial map sometimes can result into this showing a single user contribution page with the map showing all user contributions:

![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 32 09](https://user-images.githubusercontent.com/1532418/77074296-38f1ed00-6a23-11ea-94d0-b3f8763059a1.png)

norbertschuler commented on app/components/UserContributions/ContributionCard.native.js:
```diff
@@ -301,10 +300,11 @@ class ContributionCard extends React.Component {
-                contribution,
-                titleParam: plantProjectName || tpoName || treeSpecies
-              });
+              this.props.onPressSingleContribution(contribution.id);
```
You have not defined this new property in `ContributionCard.propTypes`, have you?

norbertschuler replied:
`onPressSingleContribution` still is not defined in the propTypes of `ContributionCard`, shouldn't it?

norbertschuler replied:
changed this myself

norbertschuler commented on app/components/UserContributions/ContributionCardList.native.js:
```diff
@@ -1,36 +1,43 @@
+        <ContributionCard
+          onPressSingleContribution={onPressSingleContribution}
+          contribution={item}
+          deleteContribution={deleteContribution}
```
I can not find this functionality any more in ContributionCard? How can we delete contributions like self registered trees. Maybe be lost this somehow and not in this patch, but here I recognized it first.

haideralishah replied:
not sure about it, I just reuse this component also present in develop.

norbertschuler replied:
I tested again that I was able to delete my self registered trees - single and multiple. It works, but then I get directed to the home screen and when I go back to the "Me" screen, I still see the user contribution which I just deleted.

norbertschuler commented on app/components/UserHome/AnimatedMap.js:
```diff
@@ -0,0 +1,862 @@
+import { getAllPlantProjectsSelector } from '../../selectors';
+import { markerImage } from '../../assets/index.js';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import { tree_1 } from '../../assets/index';
```
I think you can import both image assets `markerImage` and `tree_1` with one import statement.

haideralishah replied:
fixed

norbertschuler commented on app/components/UserHome/AnimatedMap.js:
```diff
@@ -0,0 +1,862 @@
+          try {
+            this.mapView.fitToSuppliedMarkers(markers.map(x => String(x.id)));
+          } catch (e) {
+            console.log(JSON.stringify(e))
```
Please replace all console.log with `debug()` or remove them.

haideralishah replied:
removed

norbertschuler commented on app/components/UserHome/panController.js:
```diff
@@ -0,0 +1,440 @@
+import React from 'react';
```
Why did you copy this code (without mentioning the initial author in a comment) into our project? Can't we use the npm module `react-native-pan-controller` directly?

haideralishah replied:
No, this is not a package this is actually an example in `react-native-maps` repo. https://github.com/react-native-community/react-native-maps/blob/master/example/examples/PanController.js

I think we can import this component like`react-native-maps/path/to/panController`
but I customise this component.

haideralishah commented:
@ankitecd please check the following issues in this pr.
- [ ] single contribution details not loading in detail page.
- [ ] check line break & `onPress` is not working on project name.
![image](https://user-images.githubusercontent.com/9020813/77653875-16ffe980-6f92-11ea-901e-5fcbb4d59bf5.png)

norbertschuler commented:
@haideralishah could you fix the eslint errors from `npm run lint:errors`:

```
/home/circleci/ttc-app/app/components/UserContributions/ContributionCard.native.js
  255:39  error  'giveeSlug' is defined but never used               no-unused-vars
  282:17  error  Strings must use singlequote                        quotes
  283:9   error  'treeCountLine' is assigned a value but never used  no-unused-vars

/home/circleci/ttc-app/app/components/UserHome/AnimatedMap.js
   10:3   error  'Text' is defined but never used                      no-unused-vars
  105:9   error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  141:17  error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  254:7   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  256:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  261:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  313:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  318:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  325:7   error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  325:24  error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  333:7   error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  333:24  error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  334:78  error  Strings must use singlequote                          quotes
  367:7   error  'lastActiveIndex' is assigned a value but never used  no-unused-vars
  370:9   error  'activeMarker' is assigned a value but never used     no-unused-vars
  410:12  error  Unexpected constant condition                         no-constant-condition
  411:13  error  Unexpected constant condition                         no-constant-condition
  423:33  error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  482:10  error  Unexpected constant condition                         no-constant-condition
  493:29  error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle

```

norbertschuler commented:
- [ ] This is what it looks like for me when I maximize the map for my user contributions:

![Simulator Screen Shot - iPhone 11 - 2020-04-07 at 18 12 19](https://user-images.githubusercontent.com/1532418/78663095-bbbbe880-78fb-11ea-82ad-8009f6b30329.png)

norbertschuler commented:
- [ ] This message on the iOS simulator occurs when I click the "get my position" button in the maximized map:
```
console.error: {"code":2,"message":"Unable to retrieve location.","PERMISSION_DENIED":1,"POSITION_UNAVAILABLE":2,"TIMEOUT":3}
```

norbertschuler commented:
- [ ] Clicking on the Project name does nothing on my iPhone X simulator:

![Simulator Screen Shot - iPhone 11 - 2020-04-07 at 18 20 43](https://user-images.githubusercontent.com/1532418/78663863-17d33c80-78fd-11ea-8ee8-673f2562a044.png)

norbertschuler commented on ios/TreecounterApp/Info.plist:
```diff
@@ -20,6 +20,10 @@
 	<string>$(MARKETING_VERSION)</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
+	<key>NSLocationWhenInUseUsageDescription</key>
```
The key `NSLocationWhenInUseUsageDescription` is already part of this file, this is a duplicate.

norbertschuler replied:
removed it

norbertschuler commented on ios/TreecounterApp/Info.plist:
```diff
@@ -20,6 +20,10 @@
 	<string>????</string>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string></string>
+	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
```
If we need this key next to `NSLocationAlwaysAndWhenInUseUsageDescription` and `NSLocationWhenInUseUsageDescription` we need to add some text to this and also do a German translation as we haven't got this resources in our LingoHub tool, but currently translate them manually in the iOS project.

norbertschuler replied:
added resource properly plus German translation

norbertschuler commented:
- [ ] I am missing user contribution details as the video or the detail card as we currently show the in the `develop` branch:

![Simulator Screen Shot - iPhone 11 - 2020-04-07 at 19 21 29](https://user-images.githubusercontent.com/1532418/78668634-32111880-7905-11ea-8e14-ae9c040a17ff.png)

sanjayradadiya commented:
@haideralishah when I delete register tree it's redirect me to userHome screen without any message 
https://www.loom.com/share/c445a85243a6483bafdfd539a5970b7c

haideralishah commented:
> https://www.loom.com/share/c445a85243a6483bafdfd539a5970b7c

fixed

norbertschuler commented:
- [ ] I found these two warnings in the browser console when I first open my "Me" screen:

```
Warning: Failed child context type: Invalid child context `virtualizedCell.cellKey` of type `number` supplied to `CellRenderer`, expected `string`.
    in CellRenderer (at VirtualizedList.js:767)
    in RCTScrollContentView (at ScrollView.js:1038)
    in RCTScrollView (at ScrollView.js:1178)
    in ScrollView (at VirtualizedList.js:1183)
    in VirtualizedList (at FlatList.js:676)
    in FlatList (at UserHome/index.native.js:617)
```

```
Warning: Failed prop type: The prop `data` is marked as required in `Carousel`, but its value is `null`.
    in Carousel (at AnimatedMap.js:431)
```

norbertschuler commented:
- [ ] I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:

![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 14](https://user-images.githubusercontent.com/1532418/79566917-85d9e980-80dd-11ea-83b4-2f498ea65b75.png)
![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 05](https://user-images.githubusercontent.com/1532418/79566929-8a060700-80dd-11ea-84cf-04b3d4bcb108.png)

norbertschuler commented:
I wonder about two further issues:

- [ ] I donated several trees in Mexico resulting into several tree icons at the same place. I see some flickering on the map where they are all located - is that a bug in the used map component?

![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 19 41 20](https://user-images.githubusercontent.com/1532418/79570736-334ffb80-80e4-11ea-8a77-35fae90bcc21.png)

- [ ] On my overview (see above) the following trees are not shown, but I can scroll to them. Is that because we cannot zoom further out of the map to show them all?

![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 19 41 51](https://user-images.githubusercontent.com/1532418/79570841-5ed2e600-80e4-11ea-92c2-d676fb49afa7.png)

norbertschuler commented:
- [ ] One more issue: When I use the button to maximize the map, I only get show the detailed map for my newest user contribution - is that's how it's wanted to be?

![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 19 50 00](https://user-images.githubusercontent.com/1532418/79571171-f3d5df00-80e4-11ea-94de-088b72220561.png)

norbertschuler commented:
- [ ] Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)

haideralishah commented:
> * [ ]  Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)

@norbertschuler Map on detail page should be in satellite mode 
this is a default requirement not a issue.

norbertschuler commented:
> > * [ ]   Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)
> 
> @norbertschuler Map on detail page should be in satellite mode
> this is a default requirement not a issue.

Yes, but if you switch it to normal mode, why does the overview map on the user contribution list page get also switched to satellite mode? That's weird ;-)

haideralishah commented:
> > > * [ ]    Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)
> > 
> > 
> > @norbertschuler Map on detail page should be in satellite mode
> > this is a default requirement not a issue.
> 
> Yes, but if you switch it to normal mode, why does the overview map on the user contribution list page get also switched to satellite mode? That's weird ;-)

Please guide me whats wrong in it.
check the gif here https://im7.ezgif.com/tmp/ezgif-7-104f32e8649d.gif

norbertschuler commented:
- [ ] Just tested the latest version: The latest bugfixes introduced a new error not showing anymore the tree icon for my first user contribution in the maximized map or in the user contribution detail page:

![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 21 15 15](https://user-images.githubusercontent.com/1532418/79578560-95aef900-80f0-11ea-843b-505a04cda626.png)

haideralishah commented:
> * [ ]  I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:
> 
> ![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 14](https://user-images.githubusercontent.com/1532418/79566917-85d9e980-80dd-11ea-83b4-2f498ea65b75.png)
> ![Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 05](https://user-images.githubusercontent.com/1532418/79566929-8a060700-80dd-11ea-84cf-04b3d4bcb108.png)

this is because of CDN problem as @sagararyal told me couple of days back.

haideralishah commented:
> * [x]  Back button is often unresponsive in the Maps/ Single Contribution Page, Improve the button performance.
> * [x]  Clicking on tree icon on map, often takes to wrong/different contribution. This has been fixed for navigation from cards, but not from icons.
> * [x]  Loading a single contribution fetches all projects related to contributions. instead it should fetch neighbors only When contribution cards are stacked as `A [B] C` B being the current view, only load projects related to A, B & C, (performance)
> * [x]  Remove Current Location Icon from the Map View
> * [x]  Change to a icon of a tree for map marker instead of image (performance)
> * [ ]  Use Apple Maps in iOS ( if it is more performant than google maps)
> * [x]  disable the marker press if detail not show
> * [x]  revert the `headerText` on outer card list
> * [x]  disable arrows keys on both ends
> * [x]  clicking on multiple card on outer list
> * [x]  Indexing issues
> * [x]  In Single Contribution, enable Satellite View by default.
> 
> Things to consider:
> 
> * [x]  Look into Map Clustering
> 
> Fixes #1958
> 
> Changes in this pull request:
> 
> * Show road map instead of satellite
> * Center Map so that more than one items are visible.
> 
> @sagararyal

single tree icon is replaced with vector icon and double trees icons are still images. Because they are not available in vector icons.

apple map logo was causing problem, it's position does not change that is why not used.

clustering package or libraries have some limitations and can not be used with functionalities that are applied in map.

norbertschuler commented:
> > * [ ]   I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:
> 
> this is because of CDN problem as @sagararyal told me couple of days back.

But how is the first screen editing the user contribution able to display the image if you can not?

norbertschuler commented:
> * [x]  One more issue: When I use the button to maximize the map, I only get show the detailed map for my newest user contribution - is that's how it's wanted to be?

@haideralishah or @sagararyal can you answer the question why we only show the first user contribution when maximizing the overview map?

norbertschuler commented:
> > > * [ ]    I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:
> > 
> > 
> > this is because of CDN problem as @sagararyal told me couple of days back.
> 
> But how is the first screen editing the user contribution able to display the image if you can not?

Fix this myself.

norbertschuler commented:
- [ ] On my Android emulator with a smaller screen I (1) was not able to scroll down the whole image for a multiple-tree registration and (2) also was not able to scroll down until the second measurement data of a single-tree registration:

![Screenshot_1587211172](https://user-images.githubusercontent.com/1532418/79637147-129caa00-81a7-11ea-8ce2-34a68b750323.png)

![Screenshot_1587211207](https://user-images.githubusercontent.com/1532418/79637154-16c8c780-81a7-11ea-8ec9-59cab5314498.png)

norbertschuler commented:
- [ ] I got this error on my Android emulator looking at a user contribution:

![Screenshot_1587295714](https://user-images.githubusercontent.com/1532418/79686732-b56b2c00-826c-11ea-93bd-ca53ab3f2cfe.png)

sanjayradadiya commented:
@haideralishah can you please check this  

- [ ]  map zoom out not working properly

>https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1995#issuecomment-601186037
>https://www.loom.com/share/e5135bf8df1c44eab8f1c4896fcb7303
![image](https://user-images.githubusercontent.com/16476120/79717274-c496bc00-82f6-11ea-8b10-7d78665b617d.png)

- [ ] after delete tree when pressing go back button throwing scrollToIndex out of range
https://www.loom.com/share/7669ec320d164bffb6068656e1888f5f

- [ ] Icon not in center 
![image](https://user-images.githubusercontent.com/16476120/79718211-23f5cb80-82f9-11ea-9396-ba6bab22f3cd.png)

- [ ] when click on specific contribution map are not moveable is this is expected behaviour ?
https://www.loom.com/share/7f5f58f70c9f4862bbed82368243f951

norbertschuler commented:
> * [x]  I got this error on my Android emulator looking at a user contribution:

Fixed this myself by adding the missing `getItemLayout` property to `FlatList` in order to enable proper `scrollToIndex` methods.

norbertschuler commented:
> * [x]  I found these two warnings in the browser console when I first open my "Me" screen:
> 
> ```
> Warning: Failed child context type: Invalid child context `virtualizedCell.cellKey` of type `number` supplied to `CellRenderer`, expected `string`.
>     in CellRenderer (at VirtualizedList.js:767)
>     in RCTScrollContentView (at ScrollView.js:1038)
>     in RCTScrollView (at ScrollView.js:1178)
>     in ScrollView (at VirtualizedList.js:1183)
>     in VirtualizedList (at FlatList.js:676)
>     in FlatList (at UserHome/index.native.js:617)
> ```

Fixed this myself.

norbertschuler commented:
- [ ] While I tried to reproduce the issues of @sanjayradadiya I also found another issue when you register your first tree:
```
TypeError: Cannot read property 'length' of null

This error is located at:
    in AnimatedViews (created by ConnectFunction)
    in ConnectFunction (at UserHome/index.native.js:278)
    in RCTView (at UserHome/index.native.js:269)
    in RCTView (at UserHome/index.native.js:526)
    in RCTScrollContentView (at ScrollView.js:1038)
    in RCTScrollView (at ScrollView.js:1151)
    in ScrollView (at UserHome/index.native.js:323)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (at react-native-safe-area-view/index.js:163)
    in SafeView (at withOrientation.js:54)
    in withOrientation (at UserHome/index.native.js:321)
    in UserHome (created by Context.Consumer)
    in withNavigation(UserHome) (at UserHome/index.js:44)
    in UserHomeContainer (created by ConnectFunction)
    in ConnectFunction (at AppDrawerNavigator.js:213)
    in Unknown (created by SceneView)
    in SceneView (created by TabNavigationView)
    in RCTView (created by ResourceSavingScene)
    in RCTView (created by ResourceSavingScene)
    in ResourceSavingScene (created by TabNavigationView)
    in RCTView (at screens.native.js:132)
    in ScreenContainer (created by TabNavigationView)
    in RCTView (created by TabNavigationView)
    in TabNavigationView (created by NavigationView)
    in NavigationView (created by Navigator)
    in Navigator (created by SceneView)
    in SceneView (created by CardContainer)
    in RCTView (created by CardContainer)
    in RCTView (created by CardContainer)
    in RCTView (created by Card)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (created by PanGestureHandler)
    in PanGestureHandler (created by Card)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (created by Card)
    in RCTView (created by Card)
    in Card (created by CardContainer)
    in CardContainer (created by CardStack)
    in RCTView (created by MaybeScreen)
    in MaybeScreen (created by CardStack)
    in RCTView (created by MaybeScreenContainer)
    in MaybeScreenContainer (created by CardStack)
    in CardStack (created by KeyboardManager)
    in KeyboardManager (created by Context.Consumer)
    in SafeAreaProviderCompat (created by StackView)
    in RCTView (created by StackView)
    in StackView (created by StackView)
    in StackView
    in Unknown (created by Navigator)
    in Navigator (created by SceneView)
    in SceneView (created by Drawer)
    in RCTView (created by ResourceSavingScene)
    in RCTView (created by ResourceSavingScene)
```
    in ResourceSavingScene (created by Drawer)
    in RCTView (at screens.native.js:132)
    in ScreenContainer (created by Drawer)
    in RCTView (at createAnimatedComponent.js:233)
    in AnimatedComponent(Component) (created by Drawer)
    in RCTView (at createAnimatedComponent.js:233)
    in AnimatedComponent(Component) (created by PanGestureHandler)
    in PanGestureHandler (created by Drawer)
    in Drawer (created by DrawerView)
    in DrawerView (created by Navigator)
    in Navigator (created by NavigationContainer)
    in NavigationContainer (at AppDrawerNavigatorContainer.js:102)
    in RCTView (at AppDrawerNavigatorContainer.js:101)
    in AppDrawerNavigatorContainer (created by ConnectFunction)
    in ConnectFunction (at TreeCounter.native.js:32)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (at src/index.tsx:85)
    in SafeAreaView (at TreeCounter.native.js:28)
    in RNCSafeAreaView (at src/index.tsx:28)
    in SafeAreaProvider (at TreeCounter.native.js:27)
    in App (created by ConnectFunction)
    in ConnectFunction (at App/index.native.js:22)
    in GlobalErrorBoundary (at App/index.native.js:21)
    in Provider (at App/index.native.js:20)
    in App (at renderApplication.js:40)
    in RCTView (at AppContainer.js:101)
    in ChildrenWrapper (at wrapRootComponent.js:9)
    in _default (at wrapRootComponent.js:8)
    in Root (at AppContainer.js:115)
    in RCTView (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:39)

norbertschuler commented:
- [ ] If you have two registered trees and delete one tree and go back to the overview you can get an exception:

```
Invariant Violation: scrollToIndex out of range: requested index 1 but maximum is 0

VirtualizedList.scrollToIndex
    index.bundle?platform=ios&dev=true&minify=false:58010:9
FlatList.scrollToIndex
    index.bundle?platform=ios&dev=true&minify=false:57636:25
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:285300:32
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:30719:21
_callTimer
    index.bundle?platform=ios&dev=true&minify=false:30636:9
Object.callTimers
    index.bundle?platform=ios&dev=true&minify=false:30843:9
MessageQueue.__callFunction
    index.bundle?platform=ios&dev=true&minify=false:2689:44
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:2402:17
MessageQueue.__guard
    index.bundle?platform=ios&dev=true&minify=false:2643:13

```

norbertschuler commented:
- [ ] Also they is a crash when you try to open and select a project for a TPO while registering trees: [register.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4508170/register.mp4.zip)
```
TypeError: Cannot read property 'type' of undefined

<unknown>
    index.bundle?platform=ios&dev=true&minify=false:209228:27
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:209256:18
Dropdown.onSelect
    index.bundle?platform=ios&dev=true&minify=false:219002:11
DropdownItem.onPress
    index.bundle?platform=ios&dev=true&minify=false:221617:11
Button.onPress
    index.bundle?platform=ios&dev=true&minify=false:221758:11
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:221975:22
_callTimer
    index.bundle?platform=ios&dev=true&minify=false:30638:9
Object.callTimers
    index.bundle?platform=ios&dev=true&minify=false:30843:9
MessageQueue.__callFunction
    index.bundle?platform=ios&dev=true&minify=false:2689:44
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:2402:17
```

norbertschuler commented:
> * [x]  If you have two registered trees and delete one tree and go back to the overview you can get an exception:
> 
> ```
> Invariant Violation: scrollToIndex out of range: requested index 1 but maximum is 0

Hopefully fixed this.

norbertschuler commented:
I extracted one issue as https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/2218 as it also happens in the develop branch.

norbertschuler commented:
> * [x]  While I tried to reproduce the issues of @sanjayradadiya I also found another issue when you register your first tree:  [register_first_tree.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4508162/register_first_tree.mp4.zip)
> 
> ```
> TypeError: Cannot read property 'length' of null

I cannot reproduce this bug anymore after fixing https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1995#issuecomment-617203533
